plugins/modules/device_credential_intent.py

```diff
@@ -0,0 +1,2323 @@
+                }
+            })
+            self.msg = "No Assignment is available"
+            self.status = "success"
+            return self
+        for site_id in self.want.get("site_id"):
+            credential_params.update({"site_id": site_id})
+            final_response.append(credential_params)
+            response = self.dnac._exec(
+                family="network_settings",
+                function='assign_device_credential_to_site_v2',
+                params=credential_params,
+            )
+            self.log(str(response))
+            if not response:
+                self.msg = "response is empty"
+                self.status = "failed"
+                return self
+
+            if not isinstance(response, dict):
+                self.msg = "response is not a dictionary"
+                self.status = "failed"
+                return self
+
+            task_id = response.get("response").get("taskId")
+            while True:
+                task_details = self.get_task_details(task_id)
+                self.log(str(task_details))
+                if task_details.get("isError") is True:
+                    self.msg = str(task_details.get("progress"))
+                    self.status = "failed"
+                    return self
+                elif task_details.get("isError") is False:
+                    self.result['changed'] = True
+                    break
+
+        self.log("Device Credential Assigned to site is Successfully")
+        result_assign_credential.update({
+            "Assign Credentials": {
+                "response": final_response,
+                "msg": "Device Credential Assigned to a site is Successfully"
+            }
+        })
+        self.msg = "Global Credential is assigned Successfully"
+        self.status = "success"
+        return self
+
+    def get_diff_merged(self, config):
+        """
+        Update or Create Global Device Credential and assign device
+        credential to a site in Cisco DNA Center based on the playbook provided.
+
+        Parameters:
+            config (list of dict) - Playbook details containing Global
+            Device Credential and assign credentials to a site information.
+
+        Returns:
+            self
+        """
+
+        if config.get("GlobalCredentialDetails") is not None:
+            self.create_device_credentials().check_return_status()
+
+        if config.get("GlobalCredentialDetails") is not None:
+            self.update_device_credentials().check_return_status()
+
+        if config.get("AssignCredentialsToSite") is not None:
+            self.assign_credentials_to_site().check_return_status()
+
+        return self
+
+    def delete_device_credential(self, config):
+        """
+        Delete Global Device Credential in Cisco DNA Center based on the playbook details.
+        Check the return value of the API with check_return_status().
+
+        Parameters:
+            None
```

Comment:
config (dict) - Playbook details containing Global Device Credential information.
and self